### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by mesfin345**
I implemented the following:
- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
- You must create a single JavaScript object with all the data from the entire form and save it in local storage. Do not create one local storage entry per input field.
 And also I did:
           -  create a "Reset" button for the form that resets the input fields and deletes the data in the local storage.

### CODE REVIEW

**Comment by mesfin345** [comment]
And this is my 3rd time repeating the module so please consider it and approve my request

File: main.js [hunk 1]
```diff
@@ -15,6 +15,14 @@ const popUp = document.querySelector(".desk-popup");
...
 const professionalArtDiv = document.querySelector(".professional-art");
 
+const error = document.getElementById("error");
+const submitForm = document.getElementById("form");
+const submitBtn = document.getElementById("btn__submit");
+
+// form data
+const email = document.getElementById("email");
+const fullName = document.getElementById("full-name");
+const textArea = document.getElementById("textarea");
```
Comment (Alfredbis29) [inline main.js:25]: - [ ] As required for this project The form data for all inputs should be stored in a single object, and saved to a single local storage registry. have a look below how it should be :point_down: 
```

const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
```

File: index.html [hunk 2]
```diff
@@ -306,7 +306,7 @@ <h3 class="best">
           I'm always interested in hearing about new projects, so if you'd like
           to chat please get in touch.
         </p>
-        <form action="https://formspree.io/f/xnqwjqwy" method="post">
+        <form id="form">
```
Comment (Alfredbis29) [inline index.html:309]: - [ ] I can see you made a lot of effort, unfortunately, you forget some but you are almost there,The contact form information should persistent to page reloads.but when we reloads the page all information disappeared, kindly fix this for your project to match all requirements :wink:

File: main.js [hunk 3]
```diff
@@ -15,6 +15,14 @@ const popUp = document.querySelector(".desk-popup");
 const closeButton2 = document.querySelector(".close-popup");
 const professionalArtDiv = document.querySelector(".professional-art");
 
+const error = document.getElementById("error");
+const submitForm = document.getElementById("form");
+const submitBtn = document.getElementById("btn__submit");
```
Comment (Alfredbis29) [inline main.js:20]: - [ ] After filling the form, and not submitting it, the entered values should be saved in the browser. but I noticed that nothing has been saved after filling the form :upside_down_face:  kindly fis this issue

File: main.js [hunk 4]
```diff
@@ -116,6 +124,16 @@ projects.map((project) => {
   professionalArtDiv.appendChild(artContainerDiv);
 });
 
+// populating form value from localStorage
+
+const localStorageData = JSON.parse(localStorage.getItem("formData"));
```
Comment (Alfredbis29) [inline main.js:129]: - [ ] When the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data.

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
## Hello @Rikol137 ,
 I'm Alfred your reviewer for this PR man_beard
 
 Great work working on your project so far! :clap: There are some issues that you still need to work on to go to the next project but you are almost there!
 
  

 ## Highlights
Great job setting up the correct linters :heavy_check_mark:
 Meaningful commit messages :heavy_check_mark:
 
 ##  Required Changes 
 
 _Check the comments under the review._
 
 ## [Optional] suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

 Cheers and Happy coding! :partying_face: :clap: :clap:

 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me  in your question so I can receive the notification.

 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 1 above)
Comment (mesfin345) [reply main.js:25]: who is @Rikol137??

File: main.js (hunk 3 above)
Comment (Qoosim) [reply main.js:20]: - [ ] This particular requirement has not been implemented. Kindly look into it and fix it.


https://user-images.githubusercontent.com/26597432/202730592-8e31a86a-14e3-40ab-b39c-5181001adda9.mp4

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @mesfin345, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi team, 
## To Highlight 🎉
GitHub flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**